### PR TITLE
Add MSnSet vignette

### PR DESCRIPTION
**Pull request by csdaw**
This PR adds a vignette describing how to make and use MSnSets.

### CODE REVIEW

**Comment by TomSmithCGAT** [comment]
@csdaw - I've made some suggested edits to the text.

I think you should also draw readers attention to the existing [MSnbase IO vignette]( https://www.bioconductor.org/packages/release/bioc/vignettes/MSnbase/inst/doc/v02-MSnbase-io.html) in the appropriate places. This also covers the basic structure of an `MSnSet` and IO.
-

File: vignettes/msnset.Rmd [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+- These underlying objects must have a specific structure. For example the sample names in the assay data (columns) must match the sample names in the phenotype data (rownames) and the assay data and feature data must have the same number of rows.
+
+Conveniently, the `MSnbase` package comes with some example MSnSets. In this
+section we will explore the `msnset` MSnSet. This data set is from an 
+iTRAQ 4-plex experiment wherein BSA and Enolase have been spiked into a background
+of _Erwinia_ proteins. See `?msnset` for more information.
+
+## Exploring an MSnSet
+
+Below we make an example list to compare to an MSnSet.
```
Comment (TomSmithCGAT) [inline vignettes/msnset.Rmd:61]: I'm unclear what the benefit of defining this list, just to explore the structure is. Personally, I'd drop this part and just go straight to `str(msnset)`

File: vignettes/msnset.Rmd [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+# access the PSM metadata
+head(fData(tmt_msnset))
+
+# access the sample metadata
+head(pData(tmt_msnset))
+```
+
+## Extracting results from an MSnSet
+
+The code below shows briefly how to save the data within an MSnSet to a csv
```
Comment (TomSmithCGAT) [inline vignettes/msnset.Rmd:324]: Should also mention `write.exprs` here

File: vignettes/msnset.Rmd [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+head(exprs(tmt_msnset))
+
+# access the PSM metadata
+head(fData(tmt_msnset))
+
+# access the sample metadata
+head(pData(tmt_msnset))
+```
+
+## Extracting results from an MSnSet
```
Comment (TomSmithCGAT) [inline vignettes/msnset.Rmd:322]: Should also mention `write.exprs` here,

**Review by TomSmithCGAT — COMMENTED** [review]
Couple of suggested updates

**Comment by csdaw** [comment]
> @csdaw - I've made some suggested edits to the text.
> 
> ## I think you should also draw readers attention to the existing [MSnbase IO vignette](https://www.bioconductor.org/packages/release/bioc/vignettes/MSnbase/inst/doc/v02-MSnbase-io.html) in the appropriate places. This also covers the basic structure of an `MSnSet` and IO.

Will do!  I might borrow Figure 3 from that vignette as well. Though, honestly I find the MSnbase vignettes can be a bit dense, especially for beginners/new labbies, which is what prompted me to write this vignette.

I agree with the other suggestions as well, thank you :)

**Comment by TomSmithCGAT** [comment]
No worries. Nice idea to write it up :grin:

**Pull request merged**